src/core/models.py

```diff
@@ -52,6 +53,9 @@ def update(self, items: dict[str, str]) -> None:
 
         if update_fields:
             self._relation_data.update(update_fields)
+            # Hack to ensure a relation-changed event -- no problem if secret-changed is unreliable
+            # see https://bugs.launchpad.net/juju/+bug/2064876
+            self._relation_data.update({"_relation_data_updated": str(uuid.uuid4())})
```

Comment:
I quite insist on this one.

Took a VERY long time of my life until the corresponding bug got filed... :-( 

With the "fix" we're safe -- as confirmed by the tests that were failing earlier.

I agree that this is not ideal. However since our goal is to deliver a stable charm -- seems necessary.

Reply:
Ignore, not the right solution -- working on a better on.

Reply:
(Also, one shortcoming: it only works when >1 nodes. If it's a single server, then it's the next `update-status` event that will populate the changes. I can add it to the commit above the workaround -- however if that's the only change left on this PR, with the 1.5 new pipeline, then I may rather do it on the next PR like [Scaling tests](https://github.com/canonical/opensearch-dashboards-operator/pull/3) ;-)  Not gonna get forgotten -- how I wish I could fortget this one :-) )